src/nvim/os/job.c

```diff
@@ -175,8 +185,8 @@ Job *job_start(JobOptions opts, int *status)
     rstream_set_stream(job->err, job->proc_stderr);
     rstream_start(job->err);
   }
-  // Save the job to the table
-  table[i] = job;
+
+  pmap_put(int)(job_map, job->id, job);
```

Comment:
Oops, I got confused and passed `-1` for the "Job table full" error... although, really, I think that will never happen.